### PR TITLE
feat: upgrade to latest Dynatrace Terraform provider

### PR DESCRIPTION
**Pull request by williamsbdev**
I am running into problems with Java and Dotnet on my machine so it's not complete as it stands but I at least wanted to get it up to you to show you we've done the work of adding the tokens.

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -128,147 +128,252 @@ func Provider() tfbridge.ProviderInfo {
...
+			"dynatrace_azure_credentials":                  {Tok: dynatraceResource(mainMod, "AzureCredentials")},
+			"dynatrace_browser_monitor":                    {Tok: dynatraceResource(mainMod, "BrowserMonitor")},
+			"dynatrace_browser_monitor_outage":             {Tok: dynatraceResource(mainMod, "BrowserMonitorOutage")},
+			"dynatrace_browser_monitor_performance":        {Tok: dynatraceResource(mainMod, "BrowserMonitorPerformance")},
+			"dynatrace_business_events_buckets":            {Tok: dynatraceResource(mainMod, "BusinessEventsBuckets")},
+			"dynatrace_business_events_metrics":            {Tok: dynatraceResource(mainMod, "BusinessEventsMetrics")},
+			"dynatrace_business_events_oneagent":           {Tok: dynatraceResource(mainMod, "BusinessEventsOneagent")},
+			"dynatrace_business_events_processing":         {Tok: dynatraceResource(mainMod, "BusinessEventsProcessing")},
+			"dynatrace_calculated_service_metric":          {Tok: dynatraceResource(mainMod, "CalculatedServiceMetric")},
+			"dynatrace_cloudfoundry_credentials":           {Tok: dynatraceResource(mainMod, "CloudfoundaryCredentials")},
```
Comment (williamsbdev) [inline provider/resources.go:158]:
```suggestion
			"dynatrace_cloudfoundry_credentials":           {Tok: dynatraceResource(mainMod, "CloudfoundryCredentials")},
```

File: provider/resources.go [hunk 2]
```diff
@@ -128,147 +128,252 @@ func Provider() tfbridge.ProviderInfo {
...
+			"dynatrace_host_naming":                        {Tok: dynatraceResource(mainMod, "HostNaming")},
+			"dynatrace_http_monitor":                       {Tok: dynatraceResource(mainMod, "HttpMonitor")},
+			"dynatrace_ibm_mq_filters":                     {Tok: dynatraceResource(mainMod, "IbmMqFilters")},
+			"dynatrace_iam_group":                          {Tok: dynatraceResource(mainMod, "IamGroup")},
+			"dynatrace_iam_permission":                     {Tok: dynatraceResource(mainMod, "IamPermission")},
+			"dynatrace_iam_policy":                         {Tok: dynatraceResource(mainMod, "IamPolicy")},
+			"dynatrace_iam_policy_bindings":                {Tok: dynatraceResource(mainMod, "IamPolicyBindings")},
+			"dynatrace_iam_user":                           {Tok: dynatraceResource(mainMod, "IamUser")},
+			"dynatrace_issue_tracking":                     {Tok: dynatraceResource(mainMod, "IssueTracking")},
+			"dynatrace_ims_bridges":                        {Tok: dynatraceResource(mainMod, "ImsBridges")},
```
Comment (williamsbdev) [inline provider/resources.go:216]:
```suggestion
			"dynatrace_iam_group":                          {Tok: dynatraceResource(mainMod, "IamGroup")},
			"dynatrace_iam_permission":                     {Tok: dynatraceResource(mainMod, "IamPermission")},
			"dynatrace_iam_policy":                         {Tok: dynatraceResource(mainMod, "IamPolicy")},
			"dynatrace_iam_policy_bindings":                {Tok: dynatraceResource(mainMod, "IamPolicyBindings")},
			"dynatrace_iam_user":                           {Tok: dynatraceResource(mainMod, "IamUser")},
			"dynatrace_ibm_mq_filters":                     {Tok: dynatraceResource(mainMod, "IbmMqFilters")},
			"dynatrace_ims_bridges":                        {Tok: dynatraceResource(mainMod, "ImsBridges")},
			"dynatrace_issue_tracking":                     {Tok: dynatraceResource(mainMod, "IssueTracking")},
```

**Comment by jaxxstorm** [comment]
@williamsbdev I used the new auto mapping functionality in the terraform providers to make this a whole lot easier. See https://github.com/lbrlabs/pulumi-dynatrace/pull/41

**Comment by williamsbdev** [comment]
> @williamsbdev I used the new auto mapping functionality in the terraform providers to make this a whole lot easier. See #41

Awesome! I made a couple of suggestions on #41 for typos and reordering the `i` resources (were slightly out of alphabetical order). Thank you so much for helping get this ready to release! My team is anxiously awaiting this new release as we're implementing Dynatrace.

**Comment by williamsbdev** [comment]
> @williamsbdev I used the new auto mapping functionality in the terraform providers to make this a whole lot easier. See #41

@jaxxstorm - How did you do this "auto mapping functionality"? Could you point me to the docs on this? I'm trying to contribute some docs on how folks can contribute to this but I'm not finding anything about this.

**Pull request closed without merging**